src/core/database.h

```diff
@@ -327,6 +327,84 @@ public :
 	void AddClassificationSelection(ClassificationSelection *classification_selection_to_add);
 	//ClassificationSelection *GetClassificationSelectionByID(long wanted_selection_id);
 
+	std::pair<std::vector<wxString>, std::vector<std::tuple<wxString, wxString, char>>> CheckSchema();
+	bool UpdateSchema(std::vector<std::tuple<wxString, wxString, char>> columns);
+
+	
+	// Description of the database schema. 
```

Comment:
One very important feature of the way you have set this up, is that the very bug prone char* used to define the type and length of the encoding is now defined in a central place. In principle, any update or to a table in other bits of the code would then be able to use this char* from the definition, rather than being re-typed. This will prevent a bunch of future bugs, I'm sure.

For this to work, I think the definitions for static_tables and dynamic_tables should be moved into their own header file database_schema.h

This header is then included in each (and only) the files it is needed. (Our core_headers is super bloated right now.)

The only other thing in that header, might be an inline convenience function that takes a wxstring (the table name) and returns this char*